src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!this.isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
Hah, no. In theory, I could've simply used `BusyItemsIterator` (and not even had to define the new `isBusy` method), but I just wanted to play it safe and deliberately iterate over all tasks in the pool, including the idle ones.